tools/clang/test/SemaHLSL/recursion-on-mesh-nodes.hlsl

```diff
@@ -0,0 +1,21 @@
+// RUN: %dxc -Tlib_6_9 %s -verify
+
+// make sure recursion isn't allowed in mesh node shaders
+
+SamplerComparisonState s;
+Texture1D t;
+
+
+[Shader("node")]
+[NodeLaunch("mesh")]
+[OutputTopology("line")]
+[NumThreads(4,1,1)]
+[NodeDispatchGrid(4,5,6)]
+[NodeIsProgramEntry]
+[NodeID("some_call_me_tim", 7)]
+[NodeLocalRootArgumentsTableIndex(13)]
+// expected-error@+2{{recursive functions are not allowed: function 'node01_mesh_dispatch' calls recursive function 'node01_mesh_dispatch'}}
+// expected-note@+1{{recursive function located here:}}
+void node01_mesh_dispatch() { 
+ node01_mesh_dispatch();
```

Comment:
This was not what was meant by recursive nodes.  We prevent this pattern from any entry point or export generally already.

What was meant by recursion was a node that outputs to an output that targets this same node, allowing it to feed itself.  Since no node output objects are allowed for mesh nodes, you are already prevented from setting up a recursive graph.

Reply:
Ah, then I will delete the recursion test, since no outputs is already tested.